### ISSUE TITLE
ResourceLoader isn't able to find resources in resw files

### ISSUE THREAD

**Issue by nickrandolph** [unoplatform/uno#11786]
### Current behavior

[BlankLocalizationTestApp01.zip](https://github.com/unoplatform/uno/files/11052936/BlankLocalizationTestApp01.zip)
Related to this issue on Extensions repo but reproducible without extensions (See attached sample): https://github.com/unoplatform/uno.extensions/issues/1307

In the attached sample the following works/doesn't work:
- XAML Localization works (elements have x:Uid specified (eg x:Uid=HelloTextBlock) and this is used to look up resource in resw file eg HelloTextBlock.Text is the key in the resw file)
- Usage of ResourceLoader has mixed result
-- WASM/GTK - ResourceLoader returns no value
-- Android - ResourceLoader returns EN resource, despite primarylanguageoverride being set to FR
-- iOS - ResourceLoader returns no value
--  Windows - the default ResourceLoader doesn't return anything but resourcemanager can be used to find the correct sub tree associated with the class library and then I can retrieve the value

### Expected behavior

There needs to be a consistent way to retrieve resources specified in resw files in code. 

### Works on UWP/WinUI

None

**Comment by nickrandolph** [comment]
@jeromelaban I think we may need to backport these to make resources accessible. It'll need an additional fix in extensions to address the use in the localization service but these fixes are enough to make resources accessible
https://github.com/unoplatform/uno/commit/3be270bead35b9ea76186f1ba797f9391b48c60f
https://github.com/unoplatform/uno/commit/dc0999068f5805979961178d42466b23c743e9bf
https://github.com/unoplatform/uno/commit/882eacabc7cacbef8f2ff4337e1f2d0592c0c3e7
https://github.com/unoplatform/uno/commit/eaaffa47a5b57c4695dfd25443229a1b0fde0c35

**Comment by Youssef1313** [comment]
For the above sample, `ResourceLoader` doesn't appear to work on WinUI. So I'm not surprised if it fails in Uno.

**Comment by nickrandolph** [comment]
@Youssef1313 yes that's correct.

For clarity there are a couple of issues:
- ResourceLoader (used on Uno) doesn't expose resources loaded from class library with the default constructor. However, you can specify a string that will pick a different ResourceLoader eg 
`new ResourceLoader("MyClassLibrary/Resources")`
However, this doesn't work on WinUI
- For WinUI you need to use ResourceManager. The MainResourceMap doesn't contain the resources from the class library. You need to retrieve the SubTree for first the class library and then the Resources file eg
manager.MainResourceMap.SubTree("MyClassLibrary").SubTree("Resources")
Long term it would be good to add an implementation of ResoruceManager to Uno to align with WinAppSdk

In meantime, in extensions we have a wrapper that handles this. It loads from a class library if one exists, otherwise will fallback to the default ResourceLoader/ResourceMap (ie for shared projects). The main part of this implementation is [here]
(https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Localization.UI/ResourceLoaderStringLocalizer.cs)

**Comment by jeromelaban** [comment]
This would mean that we'll need to implement the ResourceMap properly to align with the original behavior.

**Comment by nickrandolph** [comment]
Yes, or at least the portion of ResourceMap that allows for loading of subtrees (that seem to map to the ResourceLoaders loaded by Uno)

**Comment by nickrandolph** [comment]
[ ] Implement ResourceManager/ResourceMap to align with WinAppSdk

**Comment by nickrandolph** [comment]
Raised a discussion and closing this issue as the ability to load class library resources in Uno has been [resolved](https://github.com/unoplatform/uno/pull/11661)
